actions/ci-benchmarking/action.yml

```diff
@@ -0,0 +1,126 @@
+name: 'Go Benchmark Action'
+description: 'Run Go benchmarks and optionally publish results to GitHub Pages'
+inputs:
+  benchmarks-pr:
+    description: 'Comma-separated list of benchmarks to run on PRs (e.g., BenchmarkOne, BenchmarkTwo). Set to "from-pr" to extract from PR description.'
+    required: false
+    default: 'from-pr'
+
+  benchmarks-merge:
+    description: 'Comma-separated list of benchmarks to run on merges (e.g., BenchmarkOne, BenchmarkTwo).'
+    required: false
+    default: ''
+
+  trigger-branches:
+    # if this input did not exist, every commit to every branch (including PRs) will trigger the action for merges,
+    # this is because both PRs and pushes to branches trigger the 'push' event
+    description: 'Comma-separated list of branches for which merges will trigger this action.'
+    required: true
+
+  use-gh-pages:
+    description: 'Flag to enable publishing results to GitHub Pages (true or false)'
+    required: false
+    default: false
+
+  github-token:
+    description: 'GitHub token for authentication (GITHUB_TOKEN or a personal access token). Required if use-gh-pages is true.'
+    required: false
+
+  gh-pages-branch:
+    description: 'The GitHub Pages branch to push results to (defaults to gh-pages). Required if use-gh-pages is true.'
+    required: false
+    default: 'gh-pages'
+
+  alert-comment-cc-users:
+    description: 'Comma-separated list of users to cc on alert comments (e.g., @user1, @user2).'
+    required: false
+    default: ''
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
```

Comment:
could you please pin each action with a commit sha? like:

`actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1`

Reply:
Happy to follow our own convention, but why are we preferring commit shas over tags? The gh docs for releasing and using actions suggests using semantically tagged releases: https://docs.github.com/en/actions/creating-actions/releasing-and-maintaining-actions

Reply:
Henry mentioned this doc here:
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

The main reason I think is security hardening.

Reply:
Thanks for sharing! I thought tags were immutable